### PR TITLE
AC-1253 - Add CR task notification

### PR DESCRIPTION
**Pull request by rowellx68**
### Description

Fixes AC-1253. Add notification capability for CR tasks

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] All tests passed locally.

### CODE REVIEW

File: .octopus/deployment_process.ocl [hunk 1]
```diff
@@ -80,8 +80,19 @@ step "deploy-the-clinical-review-api" {
...
+                      secretKey: user
+                    - name: Keycloak__ClientSecret
+                      secretName: keycloak-client-secret
+                      secretKey: password
+                    - name: Aws__AccessKeyId
+                      secretName: aws-access-key-id
+                      secretKey: user
+                    - name: Aws__SecretAccessKey
+                      secretName: aws-secret-access-key
+                      secretKey: password
```
Comment (rowellx68) [inline .octopus/deployment_process.ocl:94]: Are these correct?

Comment (woodheadio) [reply .octopus/deployment_process.ocl:94]: Could you revert this change please? This file if referenced on the main branch in octopus, so putting this in the PR won't go in.

File: deploy/helm/values.yaml [hunk 2]
```diff
@@ -55,7 +61,19 @@ enviromentVariablesFromSecrets:
     - name: AideClinicalReviewService__storage__settings__accessToken
       secretName: minio-admin-user
       secretKey: password
-      
+    - name: Keycloak__ClientId
+      secretName: keycloak-client-id
```
Comment (woodheadio) [inline deploy/helm/values.yaml:65]: please could you change secretName to keycloak-clinical-review

File: deploy/helm/values.yaml [hunk 3]
```diff
@@ -55,7 +61,19 @@ enviromentVariablesFromSecrets:
...
+      secretKey: user
+    - name: Keycloak__ClientSecret
+      secretName: keycloak-client-secret
```
Comment (woodheadio) [inline deploy/helm/values.yaml:68]: keycloak-clinical-review

File: deploy/helm/values.yaml [hunk 4]
```diff
@@ -55,7 +61,19 @@ enviromentVariablesFromSecrets:
...
+      secretKey: password
+    - name: Aws__AccessKeyId
+      secretName: aws-access-key-id
```
Comment (woodheadio) [inline deploy/helm/values.yaml:71]: please could you change secretName to aws-clinical-review

File: .octopus/deployment_process.ocl (hunk 1 above)
Comment (woodheadio) [reply .octopus/deployment_process.ocl:94]: Plus the deployment process will need amending to create the secrets (if they don't exist), which we'll do in the Octopus UI.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -126,6 +126,9 @@ jobs:
       working-directory: ./src
 
     - name: Run ClinicalReview Integration Tests
+      env:
+        AWS_ACCESS_KEY_ID: "placeholder"
+        AWS_SECRET_ACCESS_KEY: "placeholder"
```
Comment (woodheadio) [inline .github/workflows/test.yml:131]: Please could you change the value to ${{ secrets.AWS_SECRET }}

File: deploy/helm/values.yaml [hunk 6]
```diff
@@ -55,7 +61,19 @@ enviromentVariablesFromSecrets:
...
+    - name: Keycloak__ClientSecret
+      secretName: keycloak-clinical-review
+      secretKey: password
+    - name: Aws__AccessKeyId
+      secretName: aws-clinical-review
+      secretKey: user
+    - name: Aws__SecretAccessKey
+      secretName: aws-clinical-review
+      secretKey: password
+
```
Comment (rowellx68) [inline deploy/helm/values.yaml:76]: @woodheadio, i've taken the liberty to update the `secretName` on line 65 to be the same as line 68. let me know if it needs changing

**Review by samrooke — APPROVED** [review]

**Review by DanielNuttall-AD — APPROVED** [review]

**Pull request merged**